appcelerator/titanium_mobile
# fix(android): themes to not use dark status/nav icons on a dark background

**JIRA:**
https://jira.appcelerator.org/browse/TIMOB-28577

**Summary:**
If `<application/>` uses Titanium's Solid Light theme (the default as of 10.1.0), then a `Ti.UI.Window` whose "theme" property is set to a dark theme might wrongly show dark status/nav bar icons on a dark background.

**Test:**
1. Set up an Android device to use "Light" mode. (Not dark mode.)
2. Build and run the below code on that device.
3. Tap on the "Theme.Titanium.Dark" button.
4. Verify status/nav bar icon light colored. _(Used to show dark icons.)_
5. Navigate back to parent window.
6. Tap on the "Theme.Titanium.Dark.Solid" button.
7. Verify status/nav bar icons are light colored. _(Used to show dark icons.)_
8. Navigate back to parent window.
9. Tap on the "Theme.Titanium.Light" button.
10. Verify status/nav bar icons are light colored. _(Used to show dark icons.)_
11. Tap on the "Theme.Titanium.Light.Solid" button.
12. Verify status/nav bar icons are **dark** colored on a white background. _(This worked before.)_
13. Set device to "Dark" mode.
14. Repeat steps 3-13. Note that only root window respects device theme, not child windows.

```javascript
const parentWindow = Ti.UI.createWindow({
	title: "Select Child Theme",
	extendSafeArea: false,
});
const tableView = Ti.UI.createTableView({
	data: [
		Ti.UI.createTableViewRow({ title: "Theme.Titanium.Dark" }),
		Ti.UI.createTableViewRow({ title: "Theme.Titanium.Dark.Solid" }),
		Ti.UI.createTableViewRow({ title: "Theme.Titanium.Light" }),
		Ti.UI.createTableViewRow({ title: "Theme.Titanium.Light.Solid" }),
	],
});
tableView.addEventListener("click", (e) => {
	const childWindow = Ti.UI.createWindow({
		title: e.row.title,
		theme: e.row.title,
	});
	const closeButton = Ti.UI.createButton({ title: "Close" });
	closeButton.addEventListener("click", () => {
		childWindow.close();
	});
	childWindow.add(closeButton);
	childWindow.open();
});
parentWindow.add(tableView);
parentWindow.open();
```
